### DISCUSSION TITLE
Inquiry about network interferograms

### DISCUSSION THREAD

**Discussion by ditafaith** [insarlab/MintPy#1288] [Q&A]
Hello,
Sorry for my poor understanding.
I don't fully understand the selection of the pair of dropped and used interferograms in routine processing. 
Is there any explanation for why MintPy still processes interferograms with red solid lines?
Any response is appreciated.
![Selection_019](https://github.com/user-attachments/assets/3fc8b983-4c1a-4f28-a130-b28a8c434d7a)

**Comment by codeautopilot** [comment]
To address your inquiry about the selection of interferogram pairs in MintPy, let's delve into the relevant parts of the codebase and documentation that might help clarify the process.

### Understanding Interferogram Selection in MintPy

1. **Interferogram Network Modification**:
   - The file `src/mintpy/modify_network.py` is crucial for modifying the network of interferograms. It allows users to remove or modify interferograms based on criteria such as coherence, baseline, and dates. This script is likely responsible for the selection of interferograms that are used or dropped during processing.

2. **Network Selection Criteria**:
   - The file `src/mintpy/utils/network.py` contains utilities for interferogram network selection. It includes functions for reading pairs list, writing pairs list, and selecting pairs based on different criteria. This might include coherence thresholds, baseline constraints, or temporal coverage.

3. **Configuration Files**:
   - The configuration files like `src/mintpy/defaults/selectNetwork.cfg` and `src/mintpy/defaults/smallbaselineApp.cfg` set default parameters for network selection and processing. These configurations might dictate which interferograms are processed or dropped.

4. **Plotting and Visualization**:
   - The file `src/mintpy/plot_network.py` is used to visualize the network of interferograms. This can help you understand which interferograms are being processed (indicated by solid lines) and which are not.

5. **FAQs and Documentation**:
   - The `docs/FAQs.md` file might contain explanations about the processing workflow, including why certain interferograms are processed despite being marked differently (e.g., with red solid lines).

### Why MintPy Processes Interferograms with Red Solid Lines

The red solid lines in your network plot might indicate interferograms that are processed under specific conditions or criteria set in the configuration files. These could be related to maintaining network connectivity, ensuring sufficient temporal coverage, or meeting specific baseline constraints.

### Steps to Investigate Further

- **Check Configuration**: Review the configuration files (`selectNetwork.cfg`, `smallbaselineApp.cfg`) to understand the criteria for selecting interferograms.
- **Modify Network**: Use `modify_network.py` to experiment with different selection criteria and see how it affects the network.
- **Visualize Network**: Use `plot_network.py` to visualize changes and understand the impact of different configurations.

If you have specific criteria or constraints in mind, you can adjust the configuration files or use the `modify_network.py` script to tailor the interferogram selection to your needs. If you need further assistance, feel free to ask!

<footer>

_Disclaimer: This comment was entirely generated using AI. Be aware that the information provided may be incorrect._

Current plan usage: 0.00%

**Have feedback or need help?**
[Discord](https://discord.gg/r72ykfvyx7)
[Documentation](https://docs.codeautopilot.com/)
[support@codeautopilot.com](mailto:support@codeautopilot.com)

</footer>

**Comment by yunjunz** [comment]
Colors (red/blue) represent the average spatial coherence; while solid and dashed lines represent the kept and excluded/dropped interferograms, as shown in the legend.

**Reply by ditafaith** [reply]
Thanks @yunjunz
If my understanding is correct, even though the solid red line did not fall with min. theresholds, MintPy tries to keep these ifgs for maintaining connection.
